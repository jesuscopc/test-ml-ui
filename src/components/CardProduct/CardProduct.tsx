import React from 'react';
import './index.scss';
import LogoML from 'assets/images/Logo_ML@2x.png.png';

const CardProduct = (): React.ReactElement => {
  return (
    <div className="cardProduct">
      <img src={LogoML} alt="imagen Product" width="180" height="180" tabIndex={0} />
      <div className="cardProduct__info">
        <div>
          <p title="Price Product" className="cardProduct__info__price" tabIndex={0}>
            $ 1980
          </p>
          <p title="Description Product" className="cardProduct__info__description" tabIndex={0}>
            Apple ipod touch 5g 16g negro Igual a nuevo Completo Unico
          </p>
        </div>
        <p title="Description Product" className="cardProduct__location" tabIndex={0}>
          Capital federal
        </p>
      </div>
    </div>
  );
};

export default CardProduct;
