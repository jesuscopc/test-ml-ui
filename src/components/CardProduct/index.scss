.cardProduct {
  grid-column-start: 2;
  grid-column-end: 11;
  display: flex;
  margin-top: 32px;
  background-color: #fff;
  padding: 20px;
  & img {
    margin-left: 16px;
    margin-right: 16px;
    border-radius: 4px;
  }
  &__info {
    width: 100%;
    display: flex;
    justify-content: space-between;
    margin-top: 10px;
    &__price {
      color: green;
      font-size: 24px;
    }
    &__description {
      font-size: 18px;
      max-width: 350px;
      color: red;
    }
  }
  &__location {
    font-size: 12px;
    margin-top: 24px;
    min-width: 71px;
    color: blue;
  }
}

@media (max-width: 500px) {
  .cardProduct {
    & img {
      width: 130px;
      height: auto;
    }
    flex-direction: column;
    & img {
      margin: auto;
    }
  }
}
