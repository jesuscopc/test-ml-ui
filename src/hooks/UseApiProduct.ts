import { useEffect, useState } from 'react';
import { SendRequest } from 'utils/Http.util';
import API from 'constants/Api';

export type TProductDetail = {
  image: string;
  title: string;
  condition: string;
  price: string;
  soldQuantity: number;
  description: string;
};

type IApiProduct = {
  data: TProductDetail;
  loading: boolean;
  error: any;
};

const UseApiProduct = (productId: string): IApiProduct => {
  const [state, setState] = useState<IApiProduct>({
    data: {
      title: '',
      condition: '',
      description: '',
      price: '',
      soldQuantity: 0,
      image: ''
    },
    loading: false,
    error: null
  });

  const getDataSearch = async () => {
    setState({ ...state, loading: true });
    const path = `${API.PRODUCT}/${productId}`;
    try {
      const result = await SendRequest(path);
      setState({ data: result.data, loading: false, error: null });
    } catch (error) {
      setState({ ...state, loading: false, error });
    }
  };

  useEffect(() => {
    getDataSearch();
  }, [productId]);

  return state;
};

export default UseApiProduct;
