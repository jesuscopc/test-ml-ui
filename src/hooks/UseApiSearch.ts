import { useEffect, useState } from 'react';
import { SendRequest } from 'utils/Http.util';
import API from 'constants/Api';

export type TProduct = {
  id: string;
  price: string;
  title: string;
  thumbnail: string;
  attributes: Array<any>;
  address: string;
};

type IApiSearch = {
  data: Array<TProduct>;
  loading: boolean;
  error: any;
};

const UseApiSearch = (searchPattern: string): IApiSearch => {
  const [state, setState] = useState<IApiSearch>({
    data: [],
    loading: false,
    error: null
  });

  const getDataSearch = async () => {
    setState({ ...state, loading: true });
    const path = `${API.SEARCH}?q=${searchPattern}`;
    try {
      const result = await SendRequest(path);
      setState({ data: result.data, loading: false, error: null });
    } catch (error) {
      setState({ data: [], loading: false, error });
    }
  };

  useEffect(() => {
    if (searchPattern.length > 1) {
      getDataSearch();
    }
  }, [searchPattern]);

  return state;
};

export default UseApiSearch;
