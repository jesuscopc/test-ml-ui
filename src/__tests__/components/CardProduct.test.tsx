import React from 'react';
import { shallow } from 'enzyme';
import CardProduct from 'components/CardProduct/CardProduct';
import { TProduct } from 'hooks/UseApiSearch';

describe('Suit test CardProduct Component', () => {
  const propsItem: TProduct = {
    id: 'MLDSHJK98',
    title: 'iPOD tOUCH',
    price: '19.9990',
    thumbnail: 'image.png',
    address: 'Distrito federal'
  };
  let wrapper = shallow(<CardProduct {...propsItem} />);
  beforeEach(() => {
    wrapper = shallow(<CardProduct {...propsItem} />);
  });
  test('should be render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should be render Title product', () => {
    expect(wrapper.find('.cardProduct__info__description').text()).toBe(propsItem.title);
  });
});
