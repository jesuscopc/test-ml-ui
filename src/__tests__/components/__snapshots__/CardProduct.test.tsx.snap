// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Suit test CardProduct Component should be render correctly 1`] = `
<div
  className="cardProduct"
>
  <Link
    tabIndex={0}
    to="/items/MLDSHJK98"
  >
    <img
      alt="iPOD tOUCH"
      height="180"
      src="image.png"
      width="180"
    />
  </Link>
  <div
    className="cardProduct__info"
  >
    <div>
      <p
        className="cardProduct__info__price"
        tabIndex={0}
        title="Price Product"
      >
        $ 19.9990
      </p>
      <Link
        tabIndex={0}
        to="/items/MLDSHJK98"
      >
        <p
          className="cardProduct__info__description"
          tabIndex={0}
          title="Description Product"
        >
          iPOD tOUCH
        </p>
      </Link>
    </div>
    <p
      className="cardProduct__location"
      tabIndex={0}
      title="Description Product"
    >
      Distrito federal
    </p>
  </div>
</div>
`;
